src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -116,15 +114,11 @@ public IDiscoveryRequest CreateDiscoveryRequest(IRequestData requestData, Discov
             return new DiscoveryRequest(requestData, discoveryCriteria, discoveryManager, loggerManager);
         }
 
-        /// <summary>
-        /// The create test run request.
-        /// </summary>
-        /// <param name="requestData">Request data.</param>
-        /// <param name="testRunCriteria"> The test run criteria.  </param>
-        /// <param name="options">Test platform options.</param>
-        /// <returns> The <see cref="ITestRunRequest"/>. </returns>
-        /// <exception cref="ArgumentNullException"> Throws if parameter is null. </exception>
-        public ITestRunRequest CreateTestRunRequest(IRequestData requestData, TestRunCriteria testRunCriteria, TestPlatformOptions options)
+        /// <inheritdoc/>
+        public ITestRunRequest CreateTestRunRequest(
```

Comment:
Yes, this should be the new normal workflow when sessions are involved. You're required to create the session first (otherwise there'll be no `ITestSession` object. You could still call `RunTests` on the `VsTestConsoleWrapper` but you'll have to provide a `TestSessionInfo` object that you won't have unless you made the `StartTestSession` call first).

You can however call `RunTests` with null for the `TestSessionInfo` parameter and you'll emulate the current behavior of TP where no testhosts are pre-started and they are spawned during the time you fulfill the request.